### DISCUSSION TITLE
Custom note titles

### DISCUSSION THREAD

**Discussion by zaaakher** [lumen-notes/lumen#246] [Ideas]
I'm just thinking out loud here,

What if we can have custom titles instead of ID's?

![lumen-custom-title](https://github.com/lumen-notes/lumen/assets/46135573/63688d19-dfaa-4ebb-8c6e-67c2dee042ba)

I'm thinking this way If I want to link another note I can just do something like @@@ to open an autocomplete popup showing me notes and filtering them as I type, so` @@@my` will show me notes that start with "my" or has "my" in the title

Overall I think ID's is not very human friendly but we still need them.

**Comment by colebemis** [comment]
Thanks for starting this discussion, @zaaakher! 

I'm curious what problem you're trying to solve here. 

If you type `[[`, you can already search by note title/body.

**Reply by zaaakher** [reply]
Thanks @colebemis, 

>If you type [[, you can already search by note title/body.

Coincidentally, I just found that out this morning 😄 The popup and autocomplete looks great 👍 

Although is there a way we can make the files names (in the repo) have custom names? So that I can tell which is which from a terminal or any git tool 
![image](https://github.com/lumen-notes/lumen/assets/46135573/59bfe7e4-26db-40a2-b4f4-400d14273cf6)


It'll be easier for humans to to navigate to the note they want (even use a url like ..../zaaakher/notes/main/journal-22th.md)

I'm assuming the ID is crucial to the inner workings of lumens to use git to store notes. But is there a way we can use something like metadata to separate what the human sees (some_name.md) from what the computer sees (123123.md) ?

**Reply by colebemis** [reply]
The id isn't actually crucial to the inner workings of Lumen. It's just a filename. You could rename those files and Lumen would still work 😄 

The only limitation right now is that you can't rename files in Lumen, but I'm open to adding that feature.

**Reply by zaaakher** [reply]
Oh I see that's good. I think it can stay the same (ID by default) until a user double-clicks on the title of the note to change the name.

**Reply by dmsrev** [reply]
This works fine in Lumen. But if I was to try to view my md files in Obsidian, for example, I can't tell what they are just by the string of numbers. And if I view them in Zettlr, it takes the title to be the first # header in the note.

**Reply by DeFrancis-unix27** [reply]
But why the need for the lumen
